src/pages/billing/NewBillingProjectWizard/GCPBillingProjectWizard/CreateTerraProjectStep.ts

```diff
@@ -0,0 +1,143 @@
+import { ButtonPrimary, Link } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import SupportRequestWrapper from 'src/components/SupportRequest'
+import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { reportErrorAndRethrow } from 'src/libs/error'
+import Events from 'src/libs/events'
+import { contactUsActive } from 'src/libs/state'
+import * as Utils from 'src/libs/utils'
+import CreateGCPBillingProject from 'src/pages/billing/CreateGCPBillingProject'
+import { BillingAccount } from 'src/pages/billing/models/BillingAccount'
+import { Step } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/Step'
+import { StepFields } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepFields'
+import { StepHeader } from 'src/pages/billing/NewBillingProjectWizard/StepWizard/StepHeader'
+
+
+interface CreateTerraProjectStepProps {
+  isActive: boolean
+  billingAccounts: Record<string, BillingAccount>
+
+  refreshed: boolean
+  setRefreshed: (boolean) => void
+  authorizeAndLoadAccounts: () => Promise<void>
+
+  onSuccess: (string) => void
+}
+
+export const CreateTerraProjectStep = ({
+  isActive,
+  billingAccounts,
+  onSuccess,
+  ...props
+}: CreateTerraProjectStepProps) => {
+  const [billingProjectName, setBillingProjectName] = useState('')
+  const [chosenBillingAccount, setChosenBillingAccount] = useState<any>()
+  const [isBusy, setIsBusy] = useState(false)
+  const [existing, setExisting] = useState<string[]>([])
+
+  const submit = _.flow(
+    reportErrorAndRethrow('Error creating billing project'),
+    Utils.withBusyState(setIsBusy)
+  )(async () => {
+    try {
+      await Ajax().Billing.createGCPProject(billingProjectName, chosenBillingAccount.accountName)
+      onSuccess(billingProjectName)
+    } catch (error: any) {
+      if (error?.status === 409) {
+        setExisting(_.concat(billingProjectName, existing))
+      } else {
+        throw error
+      }
+    }
+  })
+  return h(Step, { isActive }, [
+    h(StepHeader, {
+      title: 'STEP 4', children: [
+        'Create a Terra project to connect your Google billing account to Terra. ',
+        'Billing projects allow you to manage your workspaces and are required to create one.'
+      ]
+    }),
+    h(StepFields, { disabled: !isActive }, [
+      div([CreateGCPBillingProject({
+        billingAccounts, chosenBillingAccount, setChosenBillingAccount,
+        billingProjectName, setBillingProjectName, existing, disabled: !isActive
+      })]),
+      isActive && _.isEmpty(billingAccounts) ?
+        h(NoBillingAccounts, {
+          refreshed: props.refreshed,
+          setRefreshed: props.setRefreshed,
+          authorizeAndLoadAccounts: props.authorizeAndLoadAccounts
+        }) :
+        div({ style: { display: 'flex', flexDirection: 'column', alignItems: 'center' } }, [
+          h(ButtonPrimary, {
+            onClick: submit,
+            role: 'button',
```

Comment:
I'm pretty sure you don't need to add the button role yourself-- through some dark magic that I can't locate now in code, it gets added for you (unless you have an href, in which case this gets rendered as an anchor tag).

Reply:
It wasn't getting added. I first detected it when I switched the selector in the test to use select by role. 

I think that's why I ended up adding the 'aria-label': I assumed that the button would be selectable as a button, and was ready to gouge out my eyeballs trying to figure out why it wouldn't pick it up.

I ended up examining the DOM. and saw that the component wasn't getting the role assigned. I specified it explicitly, and the selector picked it up.

Reply:
That's weird because when I removed line 78 locally, it still had the role. And it has the role on production, even though it isn't specified explicitly in the old code. Can you try again?

<img width="703" alt="image" src="https://user-images.githubusercontent.com/484484/218129432-6f79baf6-1fd1-4654-a1df-2c50a5dd25e8.png">

Reply:
It's working now, for some reason.